LocalQueryTest.when_queryFromFollower_onStableCluster_thenReadLatestValue

http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/964/testReport/junit/com.hazelcast.cp.internal.raft.impl/LocalQueryTest/when_queryFromFollower_onStableCluster_thenReadLatestValue/

```
java.lang.AssertionError: expected:<value3> but was:<null>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.cp.internal.raft.impl.LocalQueryTest.when_queryFromFollower_onStableCluster_thenReadLatestValue(LocalQueryTest.java:130)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast/src/test/java/com/hazelcast/cp/internal/raft/impl/LocalQueryTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cp/internal/raft/impl/LocalQueryTest.java
@@ -38,10 +38,13 @@ import org.junit.runner.RunWith;
 import static com.hazelcast.cp.internal.raft.impl.RaftUtil.getCommitIndex;
 import static com.hazelcast.cp.internal.raft.impl.RaftUtil.getLeaderMember;
 import static com.hazelcast.cp.internal.raft.impl.RaftUtil.newGroupWithService;
+import static org.hamcrest.Matchers.either;
+import static org.hamcrest.Matchers.is;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
 
 @RunWith(HazelcastSerialClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
@@ -124,10 +127,12 @@ public class LocalQueryTest extends HazelcastTestSupport {
             leader.replicate(new ApplyRaftRunnable("value" + i)).get();
         }
 
-        RaftNodeImpl follower = group.getAnyFollowerNode();
+        RaftNodeImpl[] followers = group.getNodesExcept(leader.getLocalMember());
+        Object result0 = followers[0].query(new QueryRaftRunnable(), QueryPolicy.ANY_LOCAL).get();
+        Object result1 = followers[1].query(new QueryRaftRunnable(), QueryPolicy.ANY_LOCAL).get();
 
-        Object result = follower.query(new QueryRaftRunnable(), QueryPolicy.ANY_LOCAL).get();
-        assertEquals("value" + count, result);
+        String latestValue = "value" + count;
+        assertThat(latestValue, either(is(result0)).or(is(result1)));
     }
 
     @Test

--- a/hazelcast/src/test/java/com/hazelcast/cp/internal/raft/impl/testing/TestRaftMember.java
+++ b/hazelcast/src/test/java/com/hazelcast/cp/internal/raft/impl/testing/TestRaftMember.java
@@ -18,6 +18,7 @@ package com.hazelcast.cp.internal.raft.impl.testing;
 
 import com.hazelcast.core.Endpoint;
 
+import java.net.InetSocketAddress;
 import java.net.SocketAddress;
 
 public class TestRaftMember implements Endpoint {
@@ -38,7 +39,7 @@ public class TestRaftMember implements Endpoint {
 
     @Override
     public SocketAddress getSocketAddress() {
-        return null;
+        return new InetSocketAddress(port);
     }
 
     public int getPort() {

